investment-app/TRINV.Api/TRINV.Application/Commands/NewsComment/DeleteNewsCommentCommand.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace TRINV.Application.Commands.NewsComment;
+
+using Domain.Entities;
+using Interfaces;
+using MediatR;
+using Shared.Business.Exceptions;
+using Shared.Business.Extension;
+using Shared.Business.Utilities;
+
+public record DeleteNewsCommentCommand(int Id) : IRequest<OperationResult>;
+
+internal class DeleteNewsCommentCommandHandler : IRequestHandler<DeleteNewsCommentCommand, OperationResult>
+{
+    readonly IRepository<NewsComment> _newsCommentRepository;
+    readonly IUserContext _userContext;
+    readonly IUnitOfWork _unitOfWork;
+
+    public DeleteNewsCommentCommandHandler(
+        IUnitOfWork unitOfWork, 
+        IRepository<NewsComment> newsCommentRepository, IUserContext userContext)
+    {
+        _unitOfWork = unitOfWork;
+        _newsCommentRepository = newsCommentRepository;
+        _userContext = userContext;
+    }
+
+    public async Task<OperationResult> Handle(DeleteNewsCommentCommand request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult();
+
+        var newsComment = await _newsCommentRepository.GetByIdAsync(request.Id, cancellationToken);
+
+        if (newsComment == null || newsComment.NewsId != _userContext.UserId)
+            return operationResult.ReturnWithErrorMessage(
+                new NotFoundException("Comment not found"));
```

Comment:
Don't throw different types of messages. Need to be consistant.
For example you start with {object} with id {id} was not found. Continue with that.

Maybe move it in to constant

Reply:
Done.